lib/src/adaptative_builder_widget.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:flutter/widgets.dart';
+
+/// A function that creates an object of type [T].
+typedef ValueBuilder<T> = T Function(BuildContext context);
+
+abstract class AdaptativeBuilderWidget<T, B> extends StatefulWidget {
```

Comment:
Do you mean "Adaptive"? Not sure if "Adaptative" is a valid word

Reply:
Fixed
I confused the French and English word. 🙃